shell/utils/__tests__/object.test.ts

```diff
@@ -161,3 +161,69 @@ describe('fx: remove', () => {
     expect(result).toStrictEqual(expected);
   });
 });
+
+describe('fx: diff', () => {
+  it('should return an object including only the differences between two objects', () => {
+    const from = {
+      foo: 'bar',
+      baz: 'bang',
+    };
+    const to = {
+      foo:  'bar',
+      bang: 'baz'
+    };
+
+    const result = diff(from, to);
+    const expected = {
+      baz:  null,
+      bang: 'baz'
+    };
+
+    expect(result).toStrictEqual(expected);
+  });
+  it('should return an object and dot characters in object should still be respected', () => {
+    const from = {};
+    const to = { foo: { 'bar.baz': 'bang' } };
+
+    const result = diff(from, to);
+    const expected = { foo: { 'bar.baz': 'bang' } };
+
+    expect(result).toStrictEqual(expected);
+  });
+});
+
+describe('fx: definedKeys', () => {
+  it('should return an array of keys within an array', () => {
+    const obj = {
+      foo: 'bar',
+      baz: 'bang',
+    };
+
+    const result = definedKeys(obj);
+    const expected = ['"foo"', '"baz"'];
```

Comment:
Why both single and double quotes? I am not so sure to understand why do we wrap strings in quotes 🤔

Reply:
In this case, I'm testing for the presence of the double-quotes in the string. If I just wrapped them in double quotes here they would be wrapping the string but not part of the string.

The double-quotes here serve the purpose of escaping any potential `.` characters that might confuse the `set` utility function.